backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/AddVmCommand.java

```diff
@@ -505,7 +505,7 @@ protected boolean validateAddVmCommand() {
      * Check if destination storage has enough space
      */
     protected boolean validateSpaceRequirements() {
-        for (Map.Entry<Guid, List<DiskImage>> sdImageEntry : storageToDisksMap.entrySet()) {
+        for (Map.Entry<Guid, List<DiskImage>> sdImageEntry : initStorageToDisksMap().entrySet()) {
```

Comment:
We can take it offline, maybe I missed some conversation about that, sorry then :)
But what is the reason for this change?
Was there am issue to initialize the `storageToDisksMap` data member in `validate()` (line 633) as it was previously?
All the other usages AFAIU happen **after** the above initialization, so if it wasn't initialized in `validate()`  will the same code (that is now moved to `initStorageToDisksMap()` return a non-`null` result later?
Is there a difference between this usage of `storageToDisksMap` data member that you switched to calling a `initStorageToDisksMap()` method and still using the `storageToDisksMap` data member as-is in `checkTemplateImages()`?
If this is the case maybe better replace everywhere the usage of `storageToDisksMap` data member into `initStorageToDisksMap()`  method (pr probably maybe even rename it to `getStorageToDisksMap()`) and add an optimization there to also update the `storageToDisksMap` data member and return it vs. calling `ImagesHandler.buildStorageToDiskMap()` again and again? Then can also remove lines 632-633 - lazy evaluation.

Reply:
There was a discussion about that already :) [1]
Basically, this one is being initialized in the validate() and has also been accessed and used on that validation.
In that case, the storageToDisksMap contains the images as we expected.
But, as part of the cloneVm command, we're calling super.executeVmCommand(), which eventually calls canAddVm().
This one uses the storageToDisksMap, but at that point, it's no longer initialized, as its context was removed between commands.
This is way I called the initialization again for getting the data and avoiding the NPE.

[1] https://gerrit.ovirt.org/c/ovirt-engine/+/118169/12/backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/AddVmCommand.java#508

Reply:
I think this is the only disturbing part in this PR, yet not that much and we can see where it would be more appropriate to initialize that storage to disks mapping but separately

Reply:
Well, this line is the only call for recalling the initialization, for avoiding the NPE.
If there's something better you think I can do after elaborating the whole scenario above, please feel free to share :)

Reply:
this needs more thinking - the easiest would be to store the mapping that was generated at the validation phase as part of the parameters (there used to be another field that should have stored command's internal state without exposing it in the parameters but I'm not sure if it's still there) so it would be available to a new instance of the CloneVm command that tries to access `storageToDisksMap` so we'll be sure we use the mapping that was validated before. but maybe that's a waste of time because AddVm shouldn't call canAddVm from its execute method.. no need to get into this though, let's push forward this patch and think about this separately

Reply:
Ack.
Resolving for now for progressing with the patch